### PR TITLE
Pass `enabled` down to `_BackwardSyncControl`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

The strategy's `BackwardSyncControl` implementation might want to run custom code when gradient accumulation is no longer enabled. This PR allows it.

----
📚 Documentation preview 📚: https://pytorch-lightning--19577.org.readthedocs.build/en/19577/

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

File: src/lightning/fabric/CHANGELOG.md [hunk 1]
```diff
@@ -26,7 +26,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fabric now raises an error if you forget to call `fabric.backward()` when it is needed by the strategy or precision selection ([#19447](https://github.com/Lightning-AI/lightning/pull/19447), [#19493](https://github.com/Lightning-AI/lightning/pull/19493))
 
 
--
+- `_BackwardSyncControl` can now control what to do when gradient accumulation is disabled ([#TODO](https://github.com/Lightning-AI/lightning/pull/TODO))
```
Comment (carmocca) [inline src/lightning/fabric/CHANGELOG.md:29]:
```suggestion
- `_BackwardSyncControl` can now control what to do when gradient accumulation is disabled ([#19577](https://github.com/Lightning-AI/lightning/pull/19577))
```

**Review by awaelchli — APPROVED** [review]
